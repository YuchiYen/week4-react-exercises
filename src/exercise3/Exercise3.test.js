import React from 'react';
import {shallow, render} from 'enzyme';
import Exercise3 from './Exercise3';
import InfoCard from './InfoCard';

describe('Exercise3', () => {
    const wrapper = shallow(<Exercise3 />);
    const infoCard = wrapper.find('InfoCard');

    it('should render an InfoCard', () => {
        expect(infoCard).toHaveLength(1);
    });

    it('should have children', () => {
        expect(infoCard.prop('children')).toBeDefined();
    })
});

describe('InfoCard', () => {
    it('should render children', () => {
        const infoCard = render(
            <InfoCard>
                <div className="test-child">test-child</div>
            </InfoCard>
        );
        expect(infoCard.find('.test-child')).toMatchSnapshot();
    });

    it('should render a FontAwesome icon', () => {
        const infoCard = shallow(<InfoCard></InfoCard>);

        expect(infoCard.find('FontAwesomeIcon')).toHaveLength(1);
    });
});
