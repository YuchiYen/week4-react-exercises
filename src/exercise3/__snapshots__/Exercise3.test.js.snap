// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard should render children 1`] = `
<div
  class="test-child"
>
  test-child
</div>
`;
